### DISCUSSION TITLE
Use LinkedHashSet for ObservableSet default implementation.

### DISCUSSION THREAD

**Discussion by pr-1** [mobxjs/mobx.dart#867] [Ideas]
Flutter sky_engine by default uses LinkedHashSet for Set implementation. 
[https://api.flutter.dev/flutter/dart-collection/LinkedHashSet-class.html](url)

Why don't we use LinkedHashSet as default for ObservableSet constructors ?
[https://github.com/mobxjs/mobx.dart/blob/master/mobx/lib/src/api/observable_collections/observable_set.dart](url)

This causes confusion in analogy of Set and ObservableSet. Ideally they should behave same with just an observable wrapper.
But if you use the default ObservableSet it will not retain the order of iteration. on `addAll()` or any other method. 
Also I would add these things to the docs if we don't change  this behaviour.

I am aware of the alternate `ObservableSet.linkedHashSetFrom` provided separately in the api but ideally this should be the default one in my opinion.

**Comment by fzyzcjy** [comment]
LGTM, what about raising an issue or PR :)

**Reply by pr-1** [reply]
Happy to do that :)

**Comment by amondnet** [comment]
https://github.com/mobxjs/mobx.dart/pull/869
